lib/uklibparam/parser.c

```diff
@@ -597,12 +567,11 @@ static int parse_arg(struct parse_arg_ctx *ctx, char *strbuf, int scan_only)
 			 *       not allowed characters, or empty strings).
 			 */
 			if (!scan_only) {
-				ctx->ld = find_libdesc(libname, libname_len);
-				if (ctx->ld)
-					ctx->p = find_libparam(ctx->ld,
-							       paramname,
-							       paramname_len);
-				if (!ctx->ld || !ctx->p) {
+				ctx->p = find_libparam(libname,
+						       libname_len,
+						       paramname,
+						       paramname_len);
+				if (!ctx->p) {
```

Comment:
I wonder, if this is followed by an `-ENOENT` and a warning, if it would be ok to use `unlikely` here.